debugger/ptraceLib/enclave_context.c

```diff
@@ -224,8 +223,25 @@ static int _get_enclave_thread_current_ssa_info(
         return -1;
     }
 
+    // SSA is assigned to the page immediately after the tcs page.
+    // SSA address is also derived from adding tcs.ossa to enclave base address.
+    // See: host/sgx/create.c
+    // See:
+    // https://github.com/intel/linux-sgx/blob/62b116c502b09b125db9acc965694d3ecff8e698/sdk/debugger_interface/linux/se_ptrace.c#L194-L199
+    // This can be used to compute the enclave base address:
+    //    encave-base-address + tcs.ossa = tcs_addr + 4096
+    //    enclave-base-address = tcs_addr + 4096 - tcs.ossa
+    enclave_base_address = (uint8_t*)tcs_addr + 4096 - tcs.ossa;
+
+    // GS register will point to td_t. GS register value can be computed by
+    // adding tcs.gsbase to enclave base address.
+    // Note: FS register will also point to td_t upon enclave entry, but
+    // enclaves can modify FS register to implement their own threading
+    // libraries.
+    td_t_addr = (uint8_t*)enclave_base_address + tcs.gsbase;
```

Comment:
Is `tcs.gsbase` an offset relative of enclave_base_adress/TCS page?

Reply:
Yes. It is the offset of GS register value from the enclave base address.